<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.2.335">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>Dung Beetle Microbiome Figures</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1.6em;
  vertical-align: middle;
}
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<script src="dung_beetle_figs_files/libs/clipboard/clipboard.min.js"></script>
<script src="dung_beetle_figs_files/libs/quarto-html/quarto.js"></script>
<script src="dung_beetle_figs_files/libs/quarto-html/popper.min.js"></script>
<script src="dung_beetle_figs_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="dung_beetle_figs_files/libs/quarto-html/anchor.min.js"></script>
<link href="dung_beetle_figs_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="dung_beetle_figs_files/libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="dung_beetle_figs_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="dung_beetle_figs_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="dung_beetle_figs_files/libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">


</head>

<body class="fullcontent">

<div id="quarto-content" class="page-columns page-rows-contents page-layout-article">

<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Dung Beetle Microbiome Figures</h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  

</header>

<section id="dung-beetle-microbiome-response-to-drenches" class="level2">
<h2 class="anchored" data-anchor-id="dung-beetle-microbiome-response-to-drenches">Dung beetle microbiome response to drenches</h2>
<p><strong>Hypotheses</strong></p>
<ol type="1">
<li><p>The use of drenches changes the dung beetles’ microbiome composition</p></li>
<li><p>Dung beetles’ microbiome changes with diet and it is more “volatile than the microbiome of other insects</p></li>
<li><p>Species richness of gut microbiome is lower than species richness from the rest of the dung beetle (gut vs whole)</p></li>
</ol>
</section>
<section id="method-notes" class="level2">
<h2 class="anchored" data-anchor-id="method-notes">Method notes</h2>
<p><strong>Treatments</strong></p>
<ul>
<li><p>Before: Dung beetles directly from the rearing facility – from North Island, had different dung in their diet than in our study</p></li>
<li><p>Control: Dung beetles that were not subjected to any drench in their diet during the experiment</p></li>
<li><p>Drenches low: A known concentration of drenches was applied to the non-drenched dung and then fed to the beetles</p></li>
<li><p>Drenches high: 10x concentration of the drenches low treatment was applied to the non-drenched dung and then fed to the beetles</p></li>
</ul>
<p>Drenches used:</p>
<ul>
<li><p>Silver: 5 5 g/l Moxidectin</p></li>
<li><p>Blue: 2g/l abamectin, 80g/l levamisole and 45.3g/l oxfendazole</p></li>
<li><p>Red: 80g/l levamisole and 45.3g/l oxfendazole</p>
<p>Key messages from the paper:</p></li>
<li><p>Diet is the main factor in their microbiome composition</p></li>
<li><p>Due to their changing microbiome nature, they are perfect candidate for microbiome engineering studies</p></li>
</ul>
<pre><code>Known drench degraders:</code></pre>
<ul>
<li><p>Stenotrophomonas – Family Xanthomonadaceae – Class Gammaproteobacteria</p></li>
<li><p>Aeromonas – Family Aeromonadaceae – Class Gammaproteobacteria</p></li>
</ul>
<pre><code>Presumed drench degraders:</code></pre>
<ul>
<li><p>Pseudomonas – Family Pseudomonadaceae – Class Gammaproteobacteria</p>
<p><strong>Additional question: Gammaproteobacteria are known drench degraders - are they more prevalent in the gut of the dung beetle when exposed to drenches?</strong></p></li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="co"># packages</span></span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(phyloseq)</span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──
✔ dplyr     1.1.1     ✔ readr     2.1.4
✔ forcats   1.0.0     ✔ stringr   1.5.0
✔ ggplot2   3.4.2     ✔ tibble    3.2.1
✔ lubridate 1.9.2     ✔ tidyr     1.3.0
✔ purrr     1.0.1     
── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()
ℹ Use the conflicted package (&lt;http://conflicted.r-lib.org/&gt;) to force all conflicts to become errors</code></pre>
</div>
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(readxl)</span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(decontam)</span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a><span class="co"># load data</span></span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a>tax_df <span class="ot">&lt;-</span> <span class="fu">read_xlsx</span>(<span class="st">"Dung beetle microbiome paper/db_classified_16S.xlsx"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb5-6"><a href="#cb5-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(ASV,domain, phylum, class, order,family, genus)</span>
<span id="cb5-7"><a href="#cb5-7" aria-hidden="true" tabindex="-1"></a>otu_df <span class="ot">&lt;-</span> <span class="fu">read_xlsx</span>(<span class="st">"Dung beetle microbiome paper/otu_frequency_table.xlsx"</span>)</span>
<span id="cb5-8"><a href="#cb5-8" aria-hidden="true" tabindex="-1"></a>metadata_df <span class="ot">&lt;-</span> <span class="fu">read_xlsx</span>(<span class="st">"Dung beetle microbiome paper/Dung_beetles_metadata.xlsx"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb5-9"><a href="#cb5-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">separate</span>(Treatment, <span class="fu">c</span>(<span class="st">"Treatment_base"</span>, <span class="st">"level"</span>), <span class="at">remove =</span> F) <span class="sc">%&gt;%</span> </span>
<span id="cb5-10"><a href="#cb5-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">level =</span> <span class="fu">if_else</span>(<span class="fu">is.na</span>(level), <span class="st">"NA"</span>, level)) <span class="sc">%&gt;%</span> </span>
<span id="cb5-11"><a href="#cb5-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">Extraction =</span> Exctraction)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expected 2 pieces. Missing pieces filled with `NA` in 20 rows [21, 22, 23, 24,
25, 26, 27, 28, 29, 30, 31, 32, 33, 34, 35, 36, 37, 38, 39, 80].</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="co"># make phyloseq object</span></span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a>otumat <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(otu_df[<span class="sc">-</span><span class="dv">1</span>])</span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(otumat) <span class="ot">&lt;-</span> otu_df<span class="sc">$</span>ASV</span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a>taxmat<span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(tax_df[<span class="sc">-</span><span class="dv">1</span>])</span>
<span id="cb7-6"><a href="#cb7-6" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(taxmat) <span class="ot">&lt;-</span> tax_df<span class="sc">$</span>ASV</span>
<span id="cb7-7"><a href="#cb7-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-8"><a href="#cb7-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-9"><a href="#cb7-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-10"><a href="#cb7-10" aria-hidden="true" tabindex="-1"></a>OTU <span class="ot">&lt;-</span> <span class="fu">otu_table</span>(otumat, <span class="at">taxa_are_rows =</span> <span class="cn">TRUE</span>)</span>
<span id="cb7-11"><a href="#cb7-11" aria-hidden="true" tabindex="-1"></a>TAX <span class="ot">&lt;-</span> <span class="fu">tax_table</span>(taxmat)</span>
<span id="cb7-12"><a href="#cb7-12" aria-hidden="true" tabindex="-1"></a>SAM <span class="ot">&lt;-</span> <span class="fu">sample_data</span>(metadata_df)</span>
<span id="cb7-13"><a href="#cb7-13" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(SAM) <span class="ot">&lt;-</span> metadata_df<span class="sc">$</span>Samplename</span>
<span id="cb7-14"><a href="#cb7-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-15"><a href="#cb7-15" aria-hidden="true" tabindex="-1"></a>ps <span class="ot">&lt;-</span> <span class="fu">phyloseq</span>(OTU, TAX)</span>
<span id="cb7-16"><a href="#cb7-16" aria-hidden="true" tabindex="-1"></a>ps <span class="ot">&lt;-</span> <span class="fu">merge_phyloseq</span>(ps, SAM)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="co"># check library sizes</span></span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a>df <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">sample_data</span>(ps)) <span class="co"># Put sample_data into a ggplot-friendly data.frame</span></span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a>df<span class="sc">$</span>LibrarySize <span class="ot">&lt;-</span> <span class="fu">sample_sums</span>(ps)</span>
<span id="cb8-5"><a href="#cb8-5" aria-hidden="true" tabindex="-1"></a>df <span class="ot">&lt;-</span> df[<span class="fu">order</span>(df<span class="sc">$</span>LibrarySize),]</span>
<span id="cb8-6"><a href="#cb8-6" aria-hidden="true" tabindex="-1"></a>df<span class="sc">$</span>Index <span class="ot">&lt;-</span> <span class="fu">seq</span>(<span class="fu">nrow</span>(df))</span>
<span id="cb8-7"><a href="#cb8-7" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(<span class="at">data=</span>df, <span class="fu">aes</span>(<span class="at">x=</span>Index, <span class="at">y=</span>LibrarySize, <span class="at">color=</span>Treatment)) <span class="sc">+</span> <span class="fu">geom_point</span>()<span class="sc">+</span></span>
<span id="cb8-8"><a href="#cb8-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(level<span class="sc">~</span>Treatment_base)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="dung_beetle_figs_files/figure-html/unnamed-chunk-3-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>Library size looks good! The negative and wash samples have very few reads. And the others look like they have a similar spread.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="co"># clean the data using the control sample "Wash" and "Negative controls"</span></span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a><span class="fu">sample_data</span>(ps)<span class="sc">$</span>is.neg <span class="ot">&lt;-</span> <span class="fu">sample_data</span>(ps)<span class="sc">$</span>Decontamination <span class="sc">==</span> <span class="st">"Control_sample"</span></span>
<span id="cb9-4"><a href="#cb9-4" aria-hidden="true" tabindex="-1"></a>contamdf.prev <span class="ot">&lt;-</span> <span class="fu">isContaminant</span>(ps, <span class="at">method=</span><span class="st">"prevalence"</span>, <span class="at">neg=</span><span class="st">"is.neg"</span>, <span class="at">threshold =</span> <span class="fl">0.5</span>) <span class="do">## increase the threshold from 0.1 to 0.5</span></span>
<span id="cb9-5"><a href="#cb9-5" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(contamdf.prev<span class="sc">$</span>contaminant) <span class="do">## 18 contaminants found</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
FALSE  TRUE 
 1801    18 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a>ps.noncontam_bac <span class="ot">&lt;-</span> <span class="fu">prune_taxa</span>(<span class="sc">!</span>contamdf.prev<span class="sc">$</span>contaminant, ps) <span class="do">## drop these 18 contaminants</span></span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a>DATA <span class="ot">=</span> ps.noncontam_bac  <span class="sc">%&gt;%</span> </span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">subset_samples</span>(Decontamination <span class="sc">!=</span> <span class="st">"Control_sample"</span>) <span class="co"># drop these controls and just keep the real samples that are now "clean"</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a><span class="co"># selected 15,000 based on the library depths above; drops 13 samples and 60 OTUs</span></span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a><span class="do">## could also use other methods for normalisation!</span></span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a>DATA_rarefied <span class="ot">=</span> <span class="fu">rarefy_even_depth</span>(<span class="at">physeq=</span>DATA,<span class="at">sample.size=</span><span class="dv">15000</span>,<span class="at">rngseed=</span><span class="dv">7</span>,<span class="at">replace=</span><span class="cn">FALSE</span>,<span class="at">trimOTUs=</span><span class="cn">TRUE</span>,<span class="at">verbose=</span><span class="cn">TRUE</span>) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>`set.seed(7)` was used to initialize repeatable random subsampling.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Please record this for your records so others can reproduce.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Try `set.seed(7); .Random.seed` for the full vector</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>...</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>13 samples removedbecause they contained fewer reads than `sample.size`.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Up to first five removed samples are: </code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>BH2gBL1gBL2gBL3gC1g</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>...</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>6OTUs were removed because they are no longer 
present in any sample after random subsampling</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>...</code></pre>
</div>
<div class="sourceCode cell-code" id="cb23"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a>DATA_rarefied <span class="ot">&lt;-</span> <span class="fu">prune_taxa</span>(<span class="fu">taxa_sums</span>(DATA_rarefied)<span class="sc">&gt;</span><span class="dv">0</span>, DATA_rarefied)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p><strong>Hypotheses</strong></p>
<ol type="1">
<li>The use of drenches changes the dung beetles’ microbiome composition</li>
</ol>
<div class="cell">
<div class="sourceCode cell-code" id="cb24"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a><span class="co"># just the drench and the control samples - exclude the "before" samples</span></span>
<span id="cb24-2"><a href="#cb24-2" aria-hidden="true" tabindex="-1"></a>DATA_experiment <span class="ot">&lt;-</span> DATA_rarefied <span class="sc">%&gt;%</span> </span>
<span id="cb24-3"><a href="#cb24-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">subset_samples</span>(Treatment <span class="sc">!=</span> <span class="st">"Before"</span>) <span class="co"># drop these</span></span>
<span id="cb24-4"><a href="#cb24-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-5"><a href="#cb24-5" aria-hidden="true" tabindex="-1"></a>DATA_experiment <span class="ot">&lt;-</span> <span class="fu">prune_taxa</span>(<span class="fu">taxa_sums</span>(DATA_experiment)<span class="sc">&gt;</span><span class="dv">0</span>, DATA_experiment)</span>
<span id="cb24-6"><a href="#cb24-6" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb24-7"><a href="#cb24-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-8"><a href="#cb24-8" aria-hidden="true" tabindex="-1"></a>ord <span class="ot">&lt;-</span> <span class="fu">ordinate</span>(DATA_experiment, <span class="st">"MDS"</span>, <span class="st">"bray"</span>)</span>
<span id="cb24-9"><a href="#cb24-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-10"><a href="#cb24-10" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_ordination</span>(DATA_experiment, ord, <span class="at">color =</span> <span class="st">"Treatment_base"</span>, <span class="at">shape =</span> <span class="st">"level"</span>) <span class="sc">+</span></span>
<span id="cb24-11"><a href="#cb24-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="dv">3</span>)<span class="sc">+</span></span>
<span id="cb24-12"><a href="#cb24-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">c</span>( <span class="st">"blue2"</span>,</span>
<span id="cb24-13"><a href="#cb24-13" aria-hidden="true" tabindex="-1"></a>                                  <span class="st">"black"</span>, <span class="st">"red2"</span>,</span>
<span id="cb24-14"><a href="#cb24-14" aria-hidden="true" tabindex="-1"></a>                                 <span class="st">"grey70"</span>))<span class="sc">+</span></span>
<span id="cb24-15"><a href="#cb24-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_test</span>()<span class="sc">+</span></span>
<span id="cb24-16"><a href="#cb24-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span>Extraction)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="dung_beetle_figs_files/figure-html/unnamed-chunk-6-1.png" class="img-fluid" width="672"></p>
</div>
<div class="sourceCode cell-code" id="cb25"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a>psotu2veg <span class="ot">&lt;-</span> <span class="cf">function</span>(physeq) {</span>
<span id="cb25-2"><a href="#cb25-2" aria-hidden="true" tabindex="-1"></a>  OTU <span class="ot">&lt;-</span> <span class="fu">otu_table</span>(physeq)</span>
<span id="cb25-3"><a href="#cb25-3" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="fu">taxa_are_rows</span>(OTU)) {</span>
<span id="cb25-4"><a href="#cb25-4" aria-hidden="true" tabindex="-1"></a>    OTU <span class="ot">&lt;-</span> <span class="fu">t</span>(OTU)</span>
<span id="cb25-5"><a href="#cb25-5" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb25-6"><a href="#cb25-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(<span class="fu">as</span>(OTU, <span class="st">"matrix"</span>))</span>
<span id="cb25-7"><a href="#cb25-7" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb25-8"><a href="#cb25-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-9"><a href="#cb25-9" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb25-10"><a href="#cb25-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-11"><a href="#cb25-11" aria-hidden="true" tabindex="-1"></a>df_OTU <span class="ot">&lt;-</span> <span class="fu">psotu2veg</span>(DATA_experiment) <span class="do">## taxa are now columns and rows are samples, this is the correct format for vegan</span></span>
<span id="cb25-12"><a href="#cb25-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-13"><a href="#cb25-13" aria-hidden="true" tabindex="-1"></a>df_metadata <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">sample_data</span>(DATA_experiment)) <span class="sc">%&gt;%</span> </span>
<span id="cb25-14"><a href="#cb25-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">unite</span>(trt_extraction, <span class="fu">c</span>(<span class="st">"Treatment"</span>, <span class="st">"Extraction"</span>), <span class="at">sep =</span> <span class="st">"_"</span>, <span class="at">remove =</span> F) <span class="sc">%&gt;%</span> </span>
<span id="cb25-15"><a href="#cb25-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">unite</span>(trt_base_extraction, <span class="fu">c</span>(<span class="st">"Treatment_base"</span>, <span class="st">"Extraction"</span>), <span class="at">sep =</span> <span class="st">"_"</span>, <span class="at">remove =</span> F)</span>
<span id="cb25-16"><a href="#cb25-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-17"><a href="#cb25-17" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(vegan)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: permute</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: lattice</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>This is vegan 2.6-4</code></pre>
</div>
<div class="sourceCode cell-code" id="cb29"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1" aria-hidden="true" tabindex="-1"></a>dist.mat_all <span class="ot">&lt;-</span> <span class="fu">vegdist</span>(df_OTU, <span class="at">method =</span> <span class="st">"bray"</span>, <span class="at">binary=</span>F)</span>
<span id="cb29-2"><a href="#cb29-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-3"><a href="#cb29-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-4"><a href="#cb29-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-5"><a href="#cb29-5" aria-hidden="true" tabindex="-1"></a>OTU_distmat_all <span class="ot">&lt;-</span></span>
<span id="cb29-6"><a href="#cb29-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.matrix</span>(dist.mat_all, <span class="at">labels =</span> T)</span>
<span id="cb29-7"><a href="#cb29-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-8"><a href="#cb29-8" aria-hidden="true" tabindex="-1"></a>adon.results_all<span class="ot">&lt;-</span>vegan<span class="sc">::</span><span class="fu">adonis2</span>(OTU_distmat_all <span class="sc">~</span> Treatment_base<span class="sc">*</span>Extraction<span class="sc">*</span>level, <span class="at">data =</span> df_metadata, <span class="at">method=</span><span class="st">"bray"</span>,<span class="at">perm=</span><span class="dv">999</span>) <span class="do">## Treatment, extraction and their interaction are all significant</span></span>
<span id="cb29-9"><a href="#cb29-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-10"><a href="#cb29-10" aria-hidden="true" tabindex="-1"></a>adon.results_all</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Permutation test for adonis under reduced model
Terms added sequentially (first to last)
Permutation: free
Number of permutations: 999

vegan::adonis2(formula = OTU_distmat_all ~ Treatment_base * Extraction * level, data = df_metadata, permutations = 999, method = "bray")
                                Df SumOfSqs      R2      F Pr(&gt;F)    
Treatment_base                   3   1.8425 0.09991 2.4590  0.001 ***
Extraction                       1   2.0010 0.10851 8.0117  0.001 ***
level                            1   0.5218 0.02830 2.0894  0.019 *  
Treatment_base:Extraction        3   1.4534 0.07882 1.9397  0.004 ** 
Treatment_base:level             2   1.0253 0.05560 2.0525  0.005 ** 
Extraction:level                 1   0.2899 0.01572 1.1606  0.268    
Treatment_base:Extraction:level  2   0.5668 0.03074 1.1348  0.288    
Residual                        43  10.7397 0.58240                  
Total                           56  18.4404 1.00000                  
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</code></pre>
</div>
<div class="sourceCode cell-code" id="cb31"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1" aria-hidden="true" tabindex="-1"></a><span class="do">## do pairwise analysis to see which groups are different</span></span>
<span id="cb31-2"><a href="#cb31-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-3"><a href="#cb31-3" aria-hidden="true" tabindex="-1"></a><span class="do">## pairwise permanova</span></span>
<span id="cb31-4"><a href="#cb31-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(pairwiseAdonis)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: cluster</code></pre>
</div>
<div class="sourceCode cell-code" id="cb33"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1" aria-hidden="true" tabindex="-1"></a><span class="fu">pairwise.adonis</span>(OTU_distmat_all,df_metadata<span class="sc">$</span>trt_base_extraction, <span class="at">p.adjust.m =</span><span class="st">"bonferroni"</span>)<span class="co"># -- is all n.s. not sure if we want to do this</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Set of permutations &lt; 'minperm'. Generating entire set.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>'nperm' &gt;= set of all permutations: complete enumeration.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Set of permutations &lt; 'minperm'. Generating entire set.</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>                           pairs Df   SumsOfSqs    F.Model         R2   p.value
1         Blue_gut vs Blue_whole  1 0.022801997  5.6988202 0.28929754 0.0040000
2      Blue_gut vs Control_whole  1 0.008836578  2.0523698 0.18569500 0.1160000
3        Blue_gut vs Control_gut  1 0.004485654  0.8969870 0.15210937 0.5980000
4            Blue_gut vs Red_gut  1 0.003366455  0.6426510 0.05083198 0.5810000
5          Blue_gut vs Red_whole  1 0.025548777  5.6842821 0.28877264 0.0060000
6         Blue_gut vs Silver_gut  1 0.010609903  2.4041428 0.16690635 0.0700000
7       Blue_gut vs Silver_whole  1 0.020329018  5.1317696 0.28302641 0.0040000
8    Blue_whole vs Control_whole  1 0.013425102  3.8992963 0.23073720 0.0070000
9      Blue_whole vs Control_gut  1 0.007986643  2.3177692 0.20479028 0.1050000
10         Blue_whole vs Red_gut  1 0.036611194  8.5056860 0.34709030 0.0010000
11       Blue_whole vs Red_whole  1 0.003210937  0.8384452 0.04450714 0.4830000
12      Blue_whole vs Silver_gut  1 0.046858983 12.7147236 0.44279457 0.0010000
13    Blue_whole vs Silver_whole  1 0.013025580  3.8505893 0.18467532 0.0050000
14  Control_whole vs Control_gut  1 0.004686265  1.3636811 0.25424350 0.3333333
15      Control_whole vs Red_gut  1 0.012089168  2.5770204 0.18980751 0.0520000
16    Control_whole vs Red_whole  1 0.010809088  2.7196981 0.17301211 0.0190000
17   Control_whole vs Silver_gut  1 0.017622849  4.6487113 0.29706672 0.0080000
18 Control_whole vs Silver_whole  1 0.004105931  1.2244272 0.09258830 0.2850000
19        Control_gut vs Red_gut  1 0.003473021  0.6421900 0.08403219 0.7950000
20      Control_gut vs Red_whole  1 0.008868422  2.1047977 0.18953949 0.1980000
21     Control_gut vs Silver_gut  1 0.003940405  0.9867191 0.12354499 0.4250000
22   Control_gut vs Silver_whole  1 0.007112795  2.1477245 0.21164592 0.0950000
23          Red_gut vs Red_whole  1 0.041956079  8.8587916 0.35636453 0.0020000
24         Red_gut vs Silver_gut  1 0.003468230  0.7378011 0.05006182 0.5330000
25       Red_gut vs Silver_whole  1 0.029863186  6.9610196 0.31697160 0.0010000
26       Red_whole vs Silver_gut  1 0.055028388 13.3655330 0.45514355 0.0010000
27     Red_whole vs Silver_whole  1 0.006784348  1.7904783 0.09528647 0.1060000
28    Silver_gut vs Silver_whole  1 0.039570715 10.9013544 0.42087971 0.0020000
   p.adjusted sig
1       0.112    
2       1.000    
3       1.000    
4       1.000    
5       0.168    
6       1.000    
7       0.112    
8       0.196    
9       1.000    
10      0.028   .
11      1.000    
12      0.028   .
13      0.140    
14      1.000    
15      1.000    
16      0.532    
17      0.224    
18      1.000    
19      1.000    
20      1.000    
21      1.000    
22      1.000    
23      0.056    
24      1.000    
25      0.028   .
26      0.028   .
27      1.000    
28      0.056    </code></pre>
</div>
<div class="sourceCode cell-code" id="cb38"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb38-1"><a href="#cb38-1" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb38-2"><a href="#cb38-2" aria-hidden="true" tabindex="-1"></a><span class="co"># perm &lt;- how(nperm = 199)</span></span>
<span id="cb38-3"><a href="#cb38-3" aria-hidden="true" tabindex="-1"></a><span class="co"># setBlocks(perm) &lt;- with(metadata_df, sample)</span></span>
<span id="cb38-4"><a href="#cb38-4" aria-hidden="true" tabindex="-1"></a><span class="co"># blocked_adonis_away &lt;- adonis2(OTU_distmat_all_away ~ time.point*treat, data = metadata_df_away, permutations = perm) ## </span></span>
<span id="cb38-5"><a href="#cb38-5" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb38-6"><a href="#cb38-6" aria-hidden="true" tabindex="-1"></a><span class="co"># blocked_adonis_away</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<ol start="2" type="1">
<li><p>Dung beetles’ microbiome changes with diet and it is more “volatile than the microbiome of other insects</p></li>
<li><p>Species richness of gut microbiome is lower than species richness from the rest of the dung beetle (gut vs whole)</p></li>
</ol>
</section>

</main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    target: function(trigger) {
      return trigger.previousElementSibling;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->



</body></html>