<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.2.475">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>Dung Beetle Microbiome Figures</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1.6em;
  vertical-align: middle;
}
</style>


<script src="dung_beetle_figs_files/libs/clipboard/clipboard.min.js"></script>
<script src="dung_beetle_figs_files/libs/quarto-html/quarto.js"></script>
<script src="dung_beetle_figs_files/libs/quarto-html/popper.min.js"></script>
<script src="dung_beetle_figs_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="dung_beetle_figs_files/libs/quarto-html/anchor.min.js"></script>
<link href="dung_beetle_figs_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="dung_beetle_figs_files/libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="dung_beetle_figs_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="dung_beetle_figs_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="dung_beetle_figs_files/libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">


</head>

<body class="fullcontent">

<div id="quarto-content" class="page-columns page-rows-contents page-layout-article">

<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Dung Beetle Microbiome Figures</h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  

</header>

<section id="dung-beetle-microbiome-response-to-drenches" class="level2">
<h2 class="anchored" data-anchor-id="dung-beetle-microbiome-response-to-drenches">Dung beetle microbiome response to drenches</h2>
<p><strong>Hypotheses</strong> – <em>Think it might be cleaner for this paper to pose these hypotheses as questions - that we then answer with the following analyses. Think it sets this up as an exploratory study, otherwise I’d expect the hypotheses to be clearer in which direction a change would be expected.</em></p>
<ol type="1">
<li><p>The use of drenches changes the dung beetles’ microbiome composition</p></li>
<li><p>Dung beetles’ microbiome changes with diet and it is more “volatile than the microbiome of other insects</p></li>
<li><p>Species richness of gut microbiome is lower than species richness from the rest of the dung beetle (gut vs whole)</p></li>
</ol>
</section>
<section id="method-notes" class="level2">
<h2 class="anchored" data-anchor-id="method-notes">Method notes</h2>
<p><strong>Treatments</strong></p>
<ul>
<li><p>Before: Dung beetles directly from the rearing facility – from North Island, had different dung in their diet than in our study</p></li>
<li><p>Control: Dung beetles that were not subjected to any drench in their diet during the experiment</p></li>
<li><p>Drenches low: A known concentration of drenches was applied to the non-drenched dung and then fed to the beetles</p></li>
<li><p>Drenches high: 10x concentration of the drenches low treatment was applied to the non-drenched dung and then fed to the beetles</p></li>
</ul>
<p>Drenches used: (<strong>Would like to rename these for the analayses – maybe A, B, and C</strong>)</p>
<ul>
<li><p>Silver: 5 5 g/l Moxidectin</p></li>
<li><p>Blue: 2g/l abamectin, 80g/l levamisole and 45.3g/l oxfendazole</p></li>
<li><p>Red: 80g/l levamisole and 45.3g/l oxfendazole</p>
<p>Key messages from the paper:</p></li>
<li><p>Diet is the main factor in their microbiome composition</p></li>
<li><p>Due to their changing microbiome nature, they are perfect candidate for microbiome engineering studies</p></li>
</ul>
<pre><code>Known drench degraders:</code></pre>
<ul>
<li><p>Stenotrophomonas – Family Xanthomonadaceae – Class Gammaproteobacteria</p></li>
<li><p>Aeromonas – Family Aeromonadaceae – Class Gammaproteobacteria</p></li>
</ul>
<pre><code>Presumed drench degraders:</code></pre>
<ul>
<li><p>Pseudomonas – Family Pseudomonadaceae – Class Gammaproteobacteria</p>
<p><strong>Additional question: Gammaproteobacteria are known drench degraders - are they more prevalent in the gut of the dung beetle when exposed to drenches?</strong></p></li>
</ul>
<div class="cell">
<div class="cell-output cell-output-stderr">
<pre><code>── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──
✔ dplyr     1.1.1     ✔ readr     2.1.4
✔ forcats   1.0.0     ✔ stringr   1.5.0
✔ ggplot2   3.4.2     ✔ tibble    3.2.1
✔ lubridate 1.9.2     ✔ tidyr     1.3.0
✔ purrr     1.0.1     
── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()
ℹ Use the conflicted package (&lt;http://conflicted.r-lib.org/&gt;) to force all conflicts to become errors</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expected 2 pieces. Missing pieces filled with `NA` in 20 rows [21, 22, 23, 24,
25, 26, 27, 28, 29, 30, 31, 32, 33, 34, 35, 36, 37, 38, 39, 80].</code></pre>
</div>
</div>
<div class="cell">

</div>
<div class="cell">
<div class="cell-output-display">
<p><img src="dung_beetle_figs_files/figure-html/unnamed-chunk-3-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>Library size looks good! The negative and wash samples have very few reads. The others looks like they have a similar spread, but in general gut samples have fewer reads than the “whole” samples. The blue line is at 10,000 reads, and is what we’ll rarefy at.</p>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>
FALSE  TRUE 
 1801    18 </code></pre>
</div>
</div>
<p>Before we rarefy, we’ll use decontam to account for any contamination (as seen in our negative controls). Based on this, we’ll drop 18 OTUs.</p>
<div class="cell">
<div class="cell-output cell-output-stderr">
<pre><code>`set.seed(7)` was used to initialize repeatable random subsampling.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Please record this for your records so others can reproduce.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Try `set.seed(7); .Random.seed` for the full vector</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>...</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>9 samples removedbecause they contained fewer reads than `sample.size`.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Up to first five removed samples are: </code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>BL1gBL2gBL3gC1gC2g</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>...</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>16OTUs were removed because they are no longer 
present in any sample after random subsampling</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>...</code></pre>
</div>
</div>
<p>Rarefying at 10,000 reads per sample drops 9 samples and 160 OTUs.</p>
<p><strong>Hypotheses</strong></p>
<p>1. Dung beetles’ microbiome changes with diet and it is more “volatile” than the microbiome of other insects - <em>Have switched this order, because I think it helps set the stage for why we might expect changes in the microbiome – it’s flexible in the dung beetle”. We don’t have data for the comparison to the microbiome of other insects, so I think the hypothesis should be re-worded to just focus on the dung beetles and then the discussion can address the part of how this compares to other insects.</em></p>
<div class="cell">
<div class="cell-output-display">
<p><img src="dung_beetle_figs_files/figure-html/unnamed-chunk-6-1.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>Permutation test for adonis under reduced model
Terms added sequentially (first to last)
Permutation: free
Number of permutations: 999

vegan::adonis2(formula = OTU_distmat_all ~ Treatment_base + Extraction, data = df_metadata, permutations = 999, method = "bray")
               Df SumOfSqs      R2      F Pr(&gt;F)   
Treatment_base  1   0.8770 0.20886 2.9453  0.004 **
Extraction      1   0.3445 0.08204 1.1570  0.315   
Residual       10   2.9777 0.70910                 
Total          12   4.1992 1.00000                 
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</code></pre>
</div>
</div>
<p>The beetles experienced a shift in diet and environment and this affected the bacterial composition of their microbiome (R2 = .21, F= 2.95 df = 1, p = 0.005). Sampling location (gut vs exterior) was not a significant predictor of the bacterial composition (R2 = .08, F = 1.16 df = 1, p = 0.328).</p>
<p><strong>2. The use of drenches changes the dung beetles’ microbiome composition and diversity.</strong></p>
<p>For this, we subset the data to be gut and whole samples from the experimental drench application (drench exposed and control samples). - Have broken this analysis down into two parts.</p>
<ol type="1">
<li><p>Does the drench affect the bacterial composition compared to the composition of the controls? - <em>Yes,</em></p></li>
<li><p>If so, do the different drenches have different effects? (<em>have pooled the drench levels within type) - some are different, but none are different from the control group.</em></p></li>
</ol>
<div class="cell">

</div>
<p><strong>Does drench application affect the dung beetle’s bacterial composition in the gut and on the exterior?</strong> - Yes! (But extraction location is a stronger driver)</p>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>           MDS1       MDS2 Extraction Treatment_base Drench
BH1g -0.4855343  0.5707953        gut           Blue Drench
BH1w -0.9311128 -0.3513734      whole           Blue Drench
BH2g  1.1737686 -0.1014893        gut           Blue Drench
BH2w -0.2089473 -0.5911402      whole           Blue Drench
BH3g  0.8415871  0.4091332        gut           Blue Drench
BH3w -0.8988190 -0.8922860      whole           Blue Drench</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Too few points to calculate an ellipse</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 1 row containing missing values (`geom_path()`).</code></pre>
</div>
<div class="cell-output-display">
<p><img src="dung_beetle_figs_files/figure-html/unnamed-chunk-8-1.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>Permutation test for adonis under reduced model
Terms added sequentially (first to last)
Permutation: free
Number of permutations: 999

vegan::adonis2(formula = OTU_distmat_all ~ Extraction + Drench, data = df_metadata, permutations = 999, method = "bray")
           Df SumOfSqs      R2      F Pr(&gt;F)    
Extraction  1   2.3283 0.11682 8.0224  0.001 ***
Drench      1   0.7689 0.03858 2.6492  0.009 ** 
Residual   58  16.8334 0.84460                  
Total      60  19.9306 1.00000                  
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</code></pre>
</div>
</div>
<p>Composition of the bacterial communities is primarily driven by extraction location (R2 = 0.12, F = 8.02, df = 1, p = 0.001), but drench is also a key predictor of bacterial composition (R2 = 0.04, F = 2.65, p = 0.005). (*No ellipse is calculated for the control gut samples, due to too few points.)</p>
<p>Since these sample sizes are wildly different, I don’t think it makes sense to compare alpha and beta diversities at this level of resolution. Since drench did come out as a predictor (with a small influence), I think it makes sense to dig into the drench types and see if they also affect the community composition and diversity.</p>
<p><strong>Do the different drenches have different effects on composition and diversity?</strong></p>
<ul>
<li>Compostion</li>
</ul>
<div class="cell">
<div class="cell-output cell-output-stderr">
<pre><code>Too few points to calculate an ellipse</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 1 row containing missing values (`geom_path()`).</code></pre>
</div>
<div class="cell-output-display">
<p><img src="dung_beetle_figs_files/figure-html/unnamed-chunk-9-1.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>Permutation test for adonis under reduced model
Terms added sequentially (first to last)
Permutation: free
Number of permutations: 999

vegan::adonis2(formula = OTU_distmat_all ~ Extraction * Treatment_base, data = df_metadata, permutations = 999, method = "bray")
                          Df SumOfSqs      R2      F Pr(&gt;F)    
Extraction                 1   2.3283 0.11682 8.5024  0.001 ***
Treatment_base             3   1.8710 0.09387 2.2774  0.003 ** 
Extraction:Treatment_base  3   1.2175 0.06108 1.4819  0.027 *  
Residual                  53  14.5138 0.72822                  
Total                     60  19.9306 1.00000                  
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: cluster
Set of permutations &lt; 'minperm'. Generating entire set.</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>                           pairs Df   SumsOfSqs    F.Model         R2 p.value
1         Blue_gut vs Blue_whole  1 0.026289666  6.8167125 0.31245370   0.002
2      Blue_gut vs Control_whole  1 0.009752366  2.2688050 0.18492469   0.095
3        Blue_gut vs Control_gut  1 0.002533932  0.5424691 0.07192196   0.675
4            Blue_gut vs Red_gut  1 0.002029032  0.4028256 0.03005527   0.844
5          Blue_gut vs Red_whole  1 0.030391402  6.9829122 0.31765183   0.002
6         Blue_gut vs Silver_gut  1 0.008520334  2.0448373 0.12744519   0.081
7       Blue_gut vs Silver_whole  1 0.024294305  6.1477672 0.29070526   0.005
8    Blue_whole vs Control_whole  1 0.013258842  3.9707420 0.23397574   0.002
9      Blue_whole vs Control_gut  1 0.010694747  3.2289032 0.24407943   0.010
10         Blue_whole vs Red_gut  1 0.034915129  8.4764544 0.34631055   0.001
11       Blue_whole vs Red_whole  1 0.002956097  0.8097914 0.04305159   0.527
12      Blue_whole vs Silver_gut  1 0.050025117 14.4732070 0.45985803   0.001
13    Blue_whole vs Silver_whole  1 0.011055901  3.3334329 0.15625394   0.009
14  Control_whole vs Control_gut  1 0.003495710  0.9573730 0.16070389   0.473
15      Control_whole vs Red_gut  1 0.011861843  2.5564306 0.18857697   0.076
16    Control_whole vs Red_whole  1 0.011066001  2.8294657 0.17874676   0.032
17   Control_whole vs Silver_gut  1 0.017703420  4.8399122 0.28740721   0.010
18 Control_whole vs Silver_whole  1 0.004879307  1.4147668 0.09814705   0.246
19        Control_gut vs Red_gut  1 0.002874795  0.5643328 0.06589338   0.750
20      Control_gut vs Red_whole  1 0.011128241  2.7439206 0.21531212   0.054
21     Control_gut vs Silver_gut  1 0.005059222  1.3548770 0.13084434   0.227
22   Control_gut vs Silver_whole  1 0.008660980  2.5069329 0.20044346   0.041
23          Red_gut vs Red_whole  1 0.040643425  8.8669438 0.35657553   0.001
24         Red_gut vs Silver_gut  1 0.003745189  0.8461793 0.05339958   0.464
25       Red_gut vs Silver_whole  1 0.031160514  7.4046967 0.31637653   0.001
26       Red_whole vs Silver_gut  1 0.058544831 15.0358111 0.46934386   0.001
27     Red_whole vs Silver_whole  1 0.005282535  1.4163526 0.07294638   0.188
28    Silver_gut vs Silver_whole  1 0.044899104 12.6823292 0.42726867   0.001
   p.adjusted sig
1       0.056    
2       1.000    
3       1.000    
4       1.000    
5       0.056    
6       1.000    
7       0.140    
8       0.056    
9       0.280    
10      0.028   .
11      1.000    
12      0.028   .
13      0.252    
14      1.000    
15      1.000    
16      0.896    
17      0.280    
18      1.000    
19      1.000    
20      1.000    
21      1.000    
22      1.000    
23      0.028   .
24      1.000    
25      0.028   .
26      0.028   .
27      1.000    
28      0.028   .</code></pre>
</div>
</div>
<p><em>Have dropped the drench level from the analyses, due to low replication, but could put back in. It comes out as significant for the permanova for composition but with very low explanatory R2 power compared to the extraction site and treatment type).</em></p>
<p>Here we consider the effects of drench type and control on the gut and whole bacterial communities. We find that extraction type, drench types/control, and their interaction are all key predictors (extraction type: R2 = .11, df = 1, F = 8.50, p = 0.001, drench type/control: R2 = 0.09, df = 3, F = 2.28, p = 0.001, interaction: R2 = 0.06, df = 3, F = 1.48, p = 0.027).</p>
<p>Using pariwise permanova with a Bonferroni correction for multiple comparisons, we identified significant differences (p &lt; 0.05) between the following groups (Pairwise permanova table SXX):</p>
<ul>
<li><p>Blue whole vs Blue gut</p></li>
<li><p>Blue whole vs Red gut</p></li>
<li><p>Blue whole vs Silver gut</p></li>
<li><p>Red gut vs Silver whole</p></li>
<li><p>Red whole vs Silver gut</p></li>
</ul>
<p>The bacterial composition of the control communities was not significantly different from the composition of any of the drench treatment communities (p adjusted &gt; 0.05).</p>
<div class="cell">

</div>
<div class="cell">

</div>
<ul>
<li>Alpha and beta diversities</li>
</ul>
<div class="cell">
<div class="cell-output-display">
<p><img src="dung_beetle_figs_files/figure-html/unnamed-chunk-12-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>
Model selection based on AICc:

             K   AICc Delta_AICc AICcWt Cum.Wt     LL
extraction   3 196.25       0.00   0.76   0.76 -94.92
null         2 200.40       4.15   0.09   0.85 -98.10
additive     9 200.42       4.17   0.09   0.94 -89.45
interaction 15 202.57       6.32   0.03   0.98 -80.95
treatment    8 203.15       6.89   0.02   1.00 -92.19</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>Analysis of Variance Table

Response: Shannon
           Df Sum Sq Mean Sq F value  Pr(&gt;F)  
Extraction  1  8.824  8.8237  6.4882 0.01349 *
Residuals  59 80.238  1.3600                  
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</code></pre>
</div>
</div>
<p>Based on model selection, extraction location was the significant predictor of the best fit model (Table Model Selection SXS). Through comparison of this model with a null, intercept-only model, we found that bacterial alpha diversity is driven by extraction location (gut vs whole), with microbial communities in the gut being less diverse than those on the exterior of the beetles (F = 6.49, df =1, p &lt; 0.013. Drench treatment and control are not significant predictors of alpha diversity. - <em>Have dropped the amount of drench from these analyses, since it doesn’t appear to have a major impact but also the sample size within each of the groups is low and it’s not significant in the model selection.</em></p>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>
Model selection based on AICc:

            K    AICc Delta_AICc AICcWt Cum.Wt    LL
extraction  3 -134.86       0.00   0.93   0.93 70.64
additive    6 -129.28       5.58   0.06   0.99 71.42
interaction 9 -125.06       9.80   0.01   1.00 73.29
null        2 -121.32      13.54   0.00   1.00 62.76
treatment   5 -116.19      18.67   0.00   1.00 63.64</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>Analysis of Variance Table

Response: beta_all.distances
           Df  Sum Sq  Mean Sq F value    Pr(&gt;F)    
Extraction  1 0.10381 0.103812  17.381 0.0001015 ***
Residuals  59 0.35239 0.005973                      
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</code></pre>
</div>
</div>
<p>Based on model selection, extraction location was the significant predictor of the best fit model (Table Model Selection SXS). Through comparison of this model with a null, intercept-only model, we found that bacterial beta diversity is driven by extraction location (gut vs whole), with microbial communities in the gut being more variable (higher beta diversity) than those on the exterior of the beetles (F = 17.38, df =1, p &lt; 0.001). Drench treatment and control are not significant predictors of beta diversity. - <em>Have dropped the amount of drench from these analyses, since it doesn’t appear to have a major impact but also the sample size within each of the groups is low and it’s not significant in the model selection.</em></p>
<ol start="3" type="1">
<li>Species richness of gut microbiome is lower than species richness from the rest of the dung beetle (gut vs whole) – <em>Seems like the gut/external microbiome comparison ends up being the key result, think it’s worthwhile including composition and diversity in this question, and we’ve addressed this in the previous analyses</em></li>
</ol>
<ul>
<li>We’ve basically already addressed this in the analyses above. Across the board sampling location (gut vs.&nbsp;whole/external) is a key predictor of composition, alpha, and beta diversity. We see that compositions are different, beta diversity is <strong>higher</strong> in the gut than on the exoskeleton (? is this what was sampled?) and alpha diversity is <strong>lower</strong> in the gut than on the exoskeleton.</li>
</ul>
<p><strong>Taxonomic composition</strong></p>
<div class="cell">

</div>
<div class="cell">

</div>
<div class="cell">
<div class="cell-output-display">
<p><img src="dung_beetle_figs_files/figure-html/unnamed-chunk-17-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>Composition of bacterial taxa pooled at the Class level. Low abundance taxa are ones that were less than 25% of the total.</p>
<p>Finally, we’re interested in whether known drench degraders (and the presumed drench degrader) are in our samples and if they respond to the drench treatment. We see variation in the Class Gammaproteobacteria in the figure, but we can look to see if these specific genera are responding.</p>
<pre><code>Known drench degraders
-   Stenotrophomonas -- Family Xanthomonadaceae -- Class Gammaproteobacteria

-   Aeromonas -- Family Aeromonadaceae -- Class Gammaproteobacteria

Presumed drench degrader
-   Pseudomonas -- Family Pseudomonadaceae -- Class Gammaproteobacteria</code></pre>
<p><strong>Additional question: Gammaproteobacteria are known drench degraders - are they more prevalent in the gut of the dung beetle when exposed to drenches?</strong> – <em>We could try deseq2, or we can use the more qualitative approach below – <strong>how the relative abundance of Stenotrophomonas and Pseudomonas varies across the different sample types – Aeromonas was not detected</strong></em></p>
<div class="cell">
<div class="cell-output-display">
<p><img src="dung_beetle_figs_files/figure-html/unnamed-chunk-18-1.png" class="img-fluid" width="672"></p>
</div>
</div>
</section>

</main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    target: function(trigger) {
      return trigger.previousElementSibling;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->



</body></html>